CustomDictionary/CustomDictionary.cs

```diff
@@ -0,0 +1,385 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CustomDictionary
+{
+    public class CustomDictionary<TKey, TValue> : IDictionary<TKey, TValue>
```

Comment:
I would suggest to implement IReadOnlyDictionary too, it will allow users of your class use it with more specific intent